code/game/turfs/simulated/minerals.dm

```diff
@@ -218,6 +218,12 @@
 
 
 
+/turf/closed/mineral/coal
+	mineralType = /obj/item/trash/coal
```

Comment:
Do NOT use /obj/item/trash/coal as your mineralType ; it will break a lot of things. Use mineral sheets. Yes, this will require making a new object.